src/Datastore/Connection/Rest.php

```diff
@@ -43,7 +43,11 @@ public function __construct(array $config = [])
     {
         $emulatorHost = getenv('DATASTORE_EMULATOR_HOST');
 
-        $baseUri = $this->getEmulatorBaseUri(self::BASE_URI, $emulatorHost);
+        $baseUri = self::BASE_URI;
+        if ($this->emulatorEnabled($emulatorHost)) {
+            $baseUri = $this->emulatorBaseUri($emulatorHost);
```

Comment:
Would it make sense to pass down the emulator host from the client since we are checking for it there? Then here:

```php
if ((bool) $config['emulatorHost']) {
    //
}
```

Reply:
Good thinking. Addressed this in the latest commit.